### PR TITLE
doc: matter: update memory requirements table for NCS 2.8.0.

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Updated memory requirements table for the NCS 2.8.0 release.
Added nRF54L15 with TF-M memory requirements.

### CODE REVIEW

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 1]
```diff
@@ -148,99 +164,127 @@ Values are provided in kilobytes (KB).
...
-      | :ref:`Thermostat <matter_thermostat_sample>` (Debug + LTO)     |            48 |               828 |              4 |         32 |         912 |                             279 |
+      | :ref:`Thermostat <matter_thermostat_sample>` (Debug + LTO)     |            48 |               840 |              4 |         32 |         924 |                             302 |
       +----------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
-      | :ref:`Thermostat <matter_thermostat_sample>` (Release)         |            48 |               804 |              4 |         32 |         888 |                             267 |
+      | :ref:`Thermostat <matter_thermostat_sample>` (Release)         |            48 |               821 |              4 |         32 |         905 |                             290 |
       +----------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
 
-   .. tab:: nRF54L15 DK
+   .. tab:: nRF54l15 DK
```
Comment (greg-fer) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:216]:
```suggestion
   .. tab:: nRF54L15 DK
```

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 2]
```diff
@@ -148,99 +164,127 @@ Values are provided in kilobytes (KB).
...
 
-      The following table lists memory requirements for samples running on the :ref:`nRF54l15 DK <programming_board_names>` (:ref:`nrf54l15dk/nrf54l15/cpuapp <nrf54l15dk_nrf54l15>`).
+      The following table lists memory requirements for samples running on the :ref:`nRF54l15 DK <programming_board_names>` (:ref:`nrf54l15dk/nrf54l15/cpuapp <zephyr:nrf54l15dk_nrf54l15>`).
```
Comment (greg-fer) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:218]:
```suggestion
      The following table lists memory requirements for samples running on the :ref:`nRF54L15 DK <programming_board_names>` (:ref:`nrf54l15dk/nrf54l15/cpuapp <zephyr:nrf54l15dk_nrf54l15>`).
```

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 3]
```diff
@@ -148,99 +164,127 @@ Values are provided in kilobytes (KB).
...
-      | :ref:`Window Covering <matter_window_covering_sample>` (Debug + LTO) |            48 |               686 |              4 |         32 |         770 |                             178 |
+      | :ref:`Window Covering <matter_window_covering_sample>` (Debug + LTO) |            48 |               694 |              4 |         40 |         786 |                             185 |
       +----------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
-      | :ref:`Window Covering <matter_window_covering_sample>` (Release)     |            48 |               655 |              4 |         32 |         739 |                             167 |
+      | :ref:`Window Covering <matter_window_covering_sample>` (Release)     |            48 |               660 |              4 |         40 |         752 |                             177 |
       +----------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
 
+   .. tab:: nRF54l15 DK with TFM
```
Comment (greg-fer) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:266]:
```suggestion
   .. tab:: nRF54L15 DK with TF-M
```

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 4]
```diff
@@ -148,99 +164,127 @@ Values are provided in kilobytes (KB).
...
+
+      The following table lists memory requirements for samples running on the :ref:`nRF54l15 DK with TFM <programming_board_names>` (:ref:`nrf54l15dk/nrf54l15/cpuapp/ns <zephyr:nrf54l15dk_nrf54l15>`).
```
Comment (greg-fer) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:268]:
```suggestion
      The following table lists memory requirements for samples running on the :ref:`nRF54L15 DK with CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` (:ref:`nrf54l15dk/nrf54l15/cpuapp/ns <zephyr:nrf54l15dk_nrf54l15>`).
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst (hunk 3 above)
Comment (greg-fer) [reply doc/nrf/protocols/matter/getting_started/hw_requirements.rst]: I think here the usage of TF-M actually made sense -- it's shorter and gives you immediately the idea of component used. In the description, however, it's better to be consistent with most of the mentions in docs.

Comment (ArekBalysNordic) [reply doc/nrf/protocols/matter/getting_started/hw_requirements.rst]: Changed.

**Review by greg-fer — APPROVED** [review]

**Review by wiba-nordic — APPROVED** [review]

**Pull request merged**